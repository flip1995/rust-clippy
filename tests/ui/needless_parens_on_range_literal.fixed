// run-rustfix
// edition:2018

#![warn(clippy::needless_parens_on_range_literal)]
#![allow(clippy::almost_complete_letter_range)]

fn main() {
    let _ = 'a'..='z';
}
