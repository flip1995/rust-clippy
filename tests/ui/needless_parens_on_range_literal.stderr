error: needless parenthesis on range literal can be removed
  --> $DIR/needless_parens_on_range_literal.rs:8:13
   |
LL |     let _ = ('a')..=('z');
   |             ^^^^^ help: try: `'a'`
   |
   = note: `-D clippy::needless-parens-on-range-literal` implied by `-D warnings`

error: needless parenthesis on range literal can be removed
  --> $DIR/needless_parens_on_range_literal.rs:8:21
   |
LL |     let _ = ('a')..=('z');
   |                     ^^^^^ help: try: `'z'`

error: needless parenthesis on range literal can be removed
  --> $DIR/needless_parens_on_range_literal.rs:11:19
   |
LL |     let _ = (1.)..(2.);
   |                   ^^^^ help: try: `2.`

error: aborting due to 3 previous errors

